### PR TITLE
Adding new protractor tests

### PR DESCRIPTION
**Pull request by vijetmahabaleshwar-okta**
## Description:
Added the following tests - 
1. Side bar navigation
2. API URL and Life cycle tags check

Updated spec files to use "fat arrows"

## Type of Pull Request:

- [ ] Content (Documentation updates or typo-fixes)
- [ ] Blog Post
- [ ] Functional (CSS changes, JS updates, or layout modifications)
- [x] Tests added/updated

### Resolves:
* [OKTA-125840](https://oktainc.atlassian.net/browse/OKTA-125840)

## How Has This Been Tested?
- [x] I have built this locally and verified that it does not break existing functionality.
- [ ] For functional changes, I have tested against Chrome, Firefox, and Safari, and mobile.
- [x] For functional changes, I have added tests.

## Primary Reviewers:

- *Tag reviewer(s)*

- [Blog Post Google Doc](https://docs.google.com)

### CODE REVIEW

File: tests/selenium/framework/page-objects/BasePage.js [hunk 1]
```diff
@@ -54,6 +53,21 @@ class BasePage {
     })
   }
 
+  elementHasLinks(element, expectedTextArray) {
```
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/framework/page-objects/BasePage.js]: I understand that method is not named correctly. 
It's actually checking for text but the usage is to check for text in links.
Is it better to name it "elementHasText"?

File: tests/selenium/framework/page-objects/BlogPage.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ class BlogPage extends BasePage {
   
   getBlogLink(post) {
     let blogPost = this.$$blogPost.get(post);
-    let title = blogPost.element(by.css('a'));
+    let title = blogPost.element(by.css('h1 a'));
```
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/framework/page-objects/BlogPage.js]: The previous locator returned multiple links within a blog post. The new one will only return title as it's in h1 tag.

File: tests/selenium/framework/page-objects/DocsPage.js [hunk 3]
```diff
@@ -39,6 +48,33 @@ class DocsPage extends BasePage {
     return el.click();
   }
 
+  hasDeprecatedTags() {
```
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/framework/page-objects/DocsPage.js]: Are these hasTags() methods sufficient to check if the tags are working?
Is there something more I need to check like the image being used in each of them?

File: tests/selenium/spec/table-of-contents-spec.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    expect(sideBarPage.getCurrentURL()).toBe('/use_cases/mfa/');
+    expect(tocPage.level1ItemContains('Multi-Factor Authentication')).toBeTruthy();
+
+    let expectedLevel2Items = ['Introduction', 'Overview of API calls used for multi-factor authentication', 'Learn more'];
+    let expectedLevel3Items = ['Prerequisites'];
+    expect(tocPage.level2ItemsContain(expectedLevel2Items)).toBeTruthy();
+    expect(tocPage.level3ItemsVisible(expectedLevel3Items)).toBeTruthy();
+
+    tocPage.clickByLinkText('Overview of API calls used for multi-factor authentication');
+    tocPage.waitForLinkToBeDisplayed('Setting up your Okta org for MFA');
```
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/spec/table-of-contents-spec.js]: wait needed as sometimes level 3 items in table don't show up immediately after level 2 item is clicked.

File: tests/selenium/spec/table-of-contents-spec.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+describe('table of contents navigation spec', () => {
+  const tocPage = new TableOfContentsPage();
+  const sideBarPage = new SideBarPage();
+
+  beforeEach(() => {
+    tocPage.load();
+    tocPage.resizeXLarge();  // At smaller sizes, table of contents is hidden
+  });
+
+  util.itNoPhantom('has basic table of contents in the documentation page', () => {
```
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/spec/table-of-contents-spec.js]: These tests fail on phantomJS as they can't find the "Table of content" elements.

File: tests/selenium/conf.js [hunk 6]
```diff
@@ -45,7 +45,8 @@ else if (process.env.TRAVIS) {
     'version': process.env.SAUCE_BROWSER_VERSION,
     'platform': process.env.SAUCE_PLATFORM,
     'tunnel-identifier': process.env.TRAVIS_JOB_NUMBER,
-    'build': process.env.TRAVIS_BUILD_NUMBER
+    'build': process.env.TRAVIS_BUILD_NUMBER,
+    'screenResolution': "1600x1200"
```
Comment (rchild-okta) [inline tests/selenium/conf.js]: nit - single quotes

File: tests/selenium/framework/page-objects/BasePage.js (hunk 1 above)
Comment (rchild-okta) [inline tests/selenium/framework/page-objects/BasePage.js]: Yep, sounds good to me

File: tests/selenium/framework/page-objects/SideBarPage.js [hunk 7]
```diff
@@ -0,0 +1,63 @@
+'use strict';
+
+const BasePage = require('./BasePage');
+const util = require('../shared/util');
+
+class SideBarPage extends BasePage {
+  constructor() {
+    let relativeUrl = '/use_cases/authentication/';
```
Comment (rchild-okta) [inline tests/selenium/framework/page-objects/SideBarPage.js]: nit: use `const` when the variable doesn't change

File: tests/selenium/framework/page-objects/TableOfContentsPage.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+          }
+        }
+      })
+    }).then(elementList => {
+      return elementList.length == expectedTextArray.length;
+    })
+  }
+  
+  clickByLinkText(linkText) {
+    let linkItem = this.$tableOfContents.element(by.linkText(linkText));
```
Comment (rchild-okta) [inline tests/selenium/framework/page-objects/TableOfContentsPage.js]: const

File: tests/selenium/framework/page-objects/TableOfContentsPage.js [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  isTopOfPageLinkDisplayed() {
+    return this.$topOfPage.isDisplayed();
+  }
+
+  gotoTopOfPage() {
+    return this.$topOfPage.click();
+  }
+
+  waitForLinkToBeDisplayed(linkText) {
+    let linkItem = this.$tableOfContents.element(by.linkText(linkText));
```
Comment (rchild-okta) [inline tests/selenium/framework/page-objects/TableOfContentsPage.js]: const

File: tests/selenium/spec/side-bar-spec.js [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+describe('sidebar navigation spec', () => {
+  const sideBarPage = new SideBarPage();
+
+  beforeEach(() => {
+    sideBarPage.load();
+    sideBarPage.resizeXLarge(); // At smaller sizes, sidebar navigation is hidden
+  });
+
+  it('has all the links for side navigation', () => {
+    let expectedUseCases = ['Authentication', 'Multi-Factor Authentication', 'API Access Management', 'Integrate with Okta'];
```
Comment (rchild-okta) [inline tests/selenium/spec/side-bar-spec.js]: nit: nicer to break this and the large arrays below into separate lines (also replace let's with const's)

File: tests/selenium/spec/table-of-contents-spec.js [hunk 11]
```diff
@@ -0,0 +1,48 @@
...
+    expect(tocPage.level1ItemContains('Authentication')).toBeTruthy();
```
Comment (rchild-okta) [inline tests/selenium/spec/table-of-contents-spec.js]: toBe(true) - here and elsewhere

File: tests/selenium/conf.js (hunk 6 above)
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/conf.js]: Done

File: tests/selenium/framework/page-objects/SideBarPage.js (hunk 7 above)
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/framework/page-objects/SideBarPage.js]: Done

File: tests/selenium/framework/page-objects/TableOfContentsPage.js (hunk 8 above)
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/framework/page-objects/TableOfContentsPage.js]: Done

File: tests/selenium/framework/page-objects/TableOfContentsPage.js (hunk 9 above)
Comment (vijetmahabaleshwar-okta) [inline tests/selenium/framework/page-objects/TableOfContentsPage.js]: Done

**Pull request merged**